### PR TITLE
Extracted interface for leader

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: eventuate-local-java-common-util/src/main/java/io/eventuate/local/java/common/util/CommonLeaderSelector.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package io.eventuate.local.java.common.util;
+
+public interface CommonLeaderSelector {
```
Comment (cer) [inline eventuate-local-java-common-util/src/main/java/io/eventuate/local/java/common/util/CommonLeaderSelector.java]: Call this LeaderSelector or Eventuate 
Add a start() method

Move to package io.eventuate.coordination.leadership

Move it's own Gradle module: eventuate-coordination-leadership

File: eventuate-local-java-common-util/src/main/java/io/eventuate/local/java/common/util/ZkLeaderSelector.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import org.apache.curator.framework.recipes.leader.CancelLeadershipException;
+import org.apache.curator.framework.recipes.leader.LeaderSelector;
+import org.apache.curator.framework.recipes.leader.LeaderSelectorListener;
+import org.apache.curator.framework.state.ConnectionState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.UUID;
+
+public class ZkLeaderSelector implements CommonLeaderSelector {
```
Comment (cer) [inline eventuate-local-java-common-util/src/main/java/io/eventuate/local/java/common/util/ZkLeaderSelector.java]: This should be in leadership.zookeeper package.
And a corresponding ...leadership-zookeeper module

Comment (dartartem) [reply eventuate-local-java-common-util/src/main/java/io/eventuate/local/java/common/util/ZkLeaderSelector.java]: will do

File: eventuate-local-java-common-util/src/main/java/io/eventuate/local/java/common/util/CommonLeaderSelector.java (hunk 1 above)
Comment (dartartem) [reply eventuate-local-java-common-util/src/main/java/io/eventuate/local/java/common/util/CommonLeaderSelector.java]: will do

**Pull request merged**